pkg/controller/accountclaim/organizational_units.go

```diff
@@ -0,0 +1,209 @@
+	awsclient "github.com/openshift/aws-account-operator/pkg/awsclient"
+	controllerutils "github.com/openshift/aws-account-operator/pkg/controller/utils"
+)
+
+// MoveAccountToOU takes care of all the logic surrounding moving an account into an OU
+func MoveAccountToOU(r *ReconcileAccountClaim, reqLogger logr.Logger, accountClaim *awsv1alpha1.AccountClaim, account *awsv1alpha1.Account) error {
+	// aws client
+	awsClient, err := awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
+		SecretName: controllerutils.AwsSecretName,
+		NameSpace:  awsv1alpha1.AccountCrNamespace,
+		AwsRegion:  "us-east-1",
+	})
+
+	// Search for ConfigMap that holds OU mapping
+	instance := &corev1.ConfigMap{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: awsv1alpha1.AccountCrNamespace, Name: OUConfigMapName}, instance)
+	if err != nil {
+		// If we failed to retrieve the ConfigMap, simply leave the account in Root
+		unexpectedErrorMsg := fmt.Sprintf("OU: Failed to find OU mapping ConfigMap, leaving account in root")
+		reqLogger.Info(unexpectedErrorMsg)
+		accountClaim.Spec.AccountOU = "ROOT"
+		return r.specUpdate(reqLogger, accountClaim)
+	}
+
+	// Get OU ID for root and base
+	friendlyOUName := account.Spec.LegalEntity.ID
+	baseID, rootID, err := checkOUMapping(instance)
+	if err != nil {
+		invalidOUErrorMsg := fmt.Sprintf("Invalid OU ConfigMap, missing root and/or base fields: %s", instance.Data)
+		reqLogger.Error(err, invalidOUErrorMsg)
+		return err
+	}
+
+	// Log ConfigMap
+	configmapMsg := fmt.Sprintf("OU: OU mapping ConfigMap was found in namespace: %s, root: %s, base: %s", OUConfigMapName, rootID, baseID)
+	reqLogger.Info(configmapMsg)
+
+	// Create/Find account OU
+	OUID, err := CreateorFindOU(reqLogger, awsClient, account, friendlyOUName, baseID)
+	if err != nil {
+		return err
+	}
+	err = MoveAccount(reqLogger, awsClient, account, OUID, rootID)
+	if err != nil {
+		// If error was cause by the account already being inside the OU, simply update the accountclaim cr and returns
+		if err == awsv1alpha1.ErrAccAlreadyInOU {
+			// Log account already in desired location
+			accountMovedMsg := fmt.Sprintf("OU: Account %s was already in the desired OU %s", account.Name, account.Spec.LegalEntity.ID)
+			reqLogger.Info(accountMovedMsg)
+			// Update accountclaim spec
+			accountClaim.Spec.AccountOU = OUID
+			return r.specUpdate(reqLogger, accountClaim)
+		}
+		return err
+	}
+
+	// Log account moved successfully
+	accountMovedMsg := fmt.Sprintf("OU: Account %s successfully moved to OU %s", account.Name, account.Spec.LegalEntity.ID)
+	reqLogger.Info(accountMovedMsg)
+
+	// Update unclaimedAccount.Spec.AwsAccountOU
+	accountClaim.Spec.AccountOU = OUID
+	return r.specUpdate(reqLogger, accountClaim)
+}
+
+// CreateorFindOU will create or find an existing OU and return its ID
+func CreateorFindOU(reqLogger logr.Logger, client awsclient.Client, account *awsv1alpha1.Account, friendlyOUName string, baseID string) (string, error) {
```

Comment:
Please use camel case, i.e. `CreateOrFindOU`

Reply:
thanks for pointing this out